applications/FluidDynamicsApplication/custom_conditions/navier_stokes_wall_condition.cpp

```diff
@@ -214,92 +235,57 @@ void NavierStokesWallCondition<TDim,TNumNodes>::CalculateRightHandSide(VectorTyp
+    int check = Condition::Check(rCurrentProcessInfo); // Checks id > 0 and area > 0
+    if (check != 0) {
+        return check;
+    } else {
         // Checks on nodes
         // Check that the element's nodes contain all required SolutionStepData and Degrees of freedom
-        for(unsigned int i=0; i<this->GetGeometry().size(); ++i)
+        for (const auto& r_node : this->GetGeometry())
         {
-            if(this->GetGeometry()[i].SolutionStepsDataHas(VELOCITY) == false)
-                KRATOS_ERROR << "missing VELOCITY variable on solution step data for node " << this->GetGeometry()[i].Id();
-            if(this->GetGeometry()[i].SolutionStepsDataHas(PRESSURE) == false)
-                KRATOS_ERROR << "missing PRESSURE variable on solution step data for node " << this->GetGeometry()[i].Id();
-            if(this->GetGeometry()[i].SolutionStepsDataHas(MESH_VELOCITY) == false)
-                KRATOS_ERROR << "missing MESH_VELOCITY variable on solution step data for node " << this->GetGeometry()[i].Id();
-            if(this->GetGeometry()[i].SolutionStepsDataHas(ACCELERATION) == false)
-                KRATOS_ERROR << "missing ACCELERATION variable on solution step data for node " << this->GetGeometry()[i].Id();
-            if(this->GetGeometry()[i].SolutionStepsDataHas(EXTERNAL_PRESSURE) == false)
-                KRATOS_ERROR << "missing EXTERNAL_PRESSURE variable on solution step data for node " << this->GetGeometry()[i].Id();
-            if(this->GetGeometry()[i].HasDofFor(VELOCITY_X) == false ||
-               this->GetGeometry()[i].HasDofFor(VELOCITY_Y) == false ||
-               this->GetGeometry()[i].HasDofFor(VELOCITY_Z) == false)
-                KRATOS_ERROR << "missing VELOCITY component degree of freedom on node " << this->GetGeometry()[i].Id();
-            if(this->GetGeometry()[i].HasDofFor(PRESSURE) == false)
-                KRATOS_ERROR << "missing PRESSURE component degree of freedom on node " << this->GetGeometry()[i].Id();
+            if(r_node.SolutionStepsDataHas(VELOCITY) == false)
+                KRATOS_ERROR << "Missing VELOCITY variable on solution step data for node " << r_node.Id() << "." << std::endl;
+            if(r_node.SolutionStepsDataHas(PRESSURE) == false)
+                KRATOS_ERROR << "Missing PRESSURE variable on solution step data for node " << r_node.Id() << "." << std::endl;
+            if(r_node.SolutionStepsDataHas(MESH_VELOCITY) == false)
+                KRATOS_ERROR << "Missing MESH_VELOCITY variable on solution step data for node " << r_node.Id() << "." << std::endl;
+            if(r_node.SolutionStepsDataHas(ACCELERATION) == false)
+                KRATOS_ERROR << "Missing ACCELERATION variable on solution step data for node " << r_node.Id() << "." << std::endl;
+            if(r_node.SolutionStepsDataHas(EXTERNAL_PRESSURE) == false)
+                KRATOS_ERROR << "Missing EXTERNAL_PRESSURE variable on solution step data for node " << r_node.Id() << "." << std::endl;
```

Comment:
Eventhough this is old code, it would be nice if we can make it modern way of checking the vars and dofs. :D

Reply:
Done. Now we use the `KRATOS_CHECK_VARIABLE_IN_NODAL_DATA` and `KRATOS_CHECK_DOF_IN_NODE` macros.